### PR TITLE
APR-1570 - Add error logging to API clients 

### PR DESCRIPTION
**Pull request by martin-hornsby**

### CODE REVIEW

File: src/SFA.DAS.ApplyService.InternalApi/Infrastructure/ApiClientBase.cs [hunk 1]
```diff
@@ -0,0 +1,232 @@
+﻿using Microsoft.Extensions.Logging;
+using Newtonsoft.Json;
+using System;
+using System.Net;
+using System.Net.Http;
+using System.Threading.Tasks;
+
+namespace SFA.DAS.ApplyService.InternalApi.Infrastructure
+{
+    public abstract class ApiClientBase<CB>
```
Comment (MichaelYoung1981) [inline src/SFA.DAS.ApplyService.InternalApi/Infrastructure/ApiClientBase.cs:10]: This class should be reused, it seems there are 2 copies of it?

Comment (martin-hornsby) [reply src/SFA.DAS.ApplyService.InternalApi/Infrastructure/ApiClientBase.cs:10]: I can try to figure out if we can shove this somewhere else. Seems overkill to have a new class library for it. I'd imagine sticking it anywhere else due to it being Infrastructure would create problems though.

Comment (MichaelYoung1981) [reply src/SFA.DAS.ApplyService.InternalApi/Infrastructure/ApiClientBase.cs:10]: If we end up with a new project I'm sure there are other pieces of duplicate code which could go in it.

File: src/SFA.DAS.ApplyService.InternalApi/Infrastructure/AssessorServiceApiClient.cs [hunk 2]
```diff
@@ -16,27 +16,28 @@
...
+            }
+            _httpClient.DefaultRequestHeaders.Authorization = new AuthenticationHeaderValue("Bearer", GetToken());
         }
 
         public async Task<IEnumerable<Types.OrganisationSearchResult>> SearchOrgansiation(string searchTerm)
         {
             _logger.LogInformation($"Searching EPAO Register. Search Term: {searchTerm}");
             var apiResponse =
-                await Get<IEnumerable<OrganisationSummary>>($"/api/ao/assessment-organisations/search/{searchTerm}");
+                await Get<List<OrganisationSummary>>($"/api/ao/assessment-organisations/search/{searchTerm}");
```
Comment (MarkFCain) [inline src/SFA.DAS.ApplyService.InternalApi/Infrastructure/AssessorServiceApiClient.cs:40]: There's a lot of this replacing IEnumerable with List, and the reason isn;'t clear in the code.  What's it for?

Comment (martin-hornsby) [reply src/SFA.DAS.ApplyService.InternalApi/Infrastructure/AssessorServiceApiClient.cs:40]: Needed for generics to work correctly

Comment (MarkFCain) [reply src/SFA.DAS.ApplyService.InternalApi/Infrastructure/AssessorServiceApiClient.cs:40]: ok

File: src/SFA.DAS.ApplyService.InternalApi/Infrastructure/ApiClientBase.cs (hunk 1 above)
Comment (martin-hornsby) [reply src/SFA.DAS.ApplyService.InternalApi/Infrastructure/ApiClientBase.cs]: Done

File: src/SFA.DAS.ApplyService.InternalApi/Startup.cs [hunk 3]
```diff
@@ -215,7 +175,56 @@ public void Configure(IApplicationBuilder app, IHostingEnvironment env)
...
         }
-        
+
+        private void ConfigHttpClients(IServiceCollection services)
+        {
+            var acceptHeaderName = "Accept";
+            var acceptHeaderValue = "application/json";
+            var handlerLifeTime = TimeSpan.FromMinutes(5);
+
+            services.AddHttpClient<AssessorServiceApiClient, AssessorServiceApiClient>(config =>
```
Comment (David-B-Read) [inline src/SFA.DAS.ApplyService.InternalApi/Startup.cs:185]: Some of these clients are EPAO-related, are they needed in Apply now it's RoATP-centric?

Comment (martin-hornsby) [reply src/SFA.DAS.ApplyService.InternalApi/Startup.cs:185]: It should be used for RoEPAO checks only for gateway 😺

**Pull request merged**